keras_nlp/models/gpt_neox/gpt_neox_backbone.py

```diff
@@ -0,0 +1,159 @@
+        max_sequence_length=512,
+        **kwargs,
+    ):
+        # Inputs
+        token_ids = keras.Input(shape=(None,), dtype="int32", name="token_ids")
+        padding_mask = keras.Input(
+            shape=(None,), dtype="int32", name="padding_mask"
+        )
+
+        # Embed tokens, positions.
+        token_embedding = keras.layers.Embedding(
+            input_dim=vocabulary_size,
+            output_dim=hidden_dim,
+            embeddings_initializer=_gpt_neox_kernel_initializer(stddev=0.01),
+            name="token_embedding",
+        )(token_ids)
+
+        x = keras.layers.Dropout(
+            dropout,
+            name="embeddings_dropout",
+        )(token_embedding)
+
+        # Apply successive transformer decoder blocks.
+        for i in range(num_layers):
+            x = GPTNeoXDecoder(
+                intermediate_dim=intermediate_dim,
+                num_heads=num_heads,
+                dropout=dropout,
+                rotary_pct=rotary_pct,
+                rotary_emb_base=rotary_emb_base,
+                layer_norm_epsilon=layer_norm_epsilon,
+                activation=lambda x: keras.activations.gelu(
+                    x, approximate=True
+                ),
+                kernel_initializer=_gpt_neox_kernel_initializer(stddev=0.02),
+                name=f"transformer_layer_{i}",
+            )(x, decoder_padding_mask=padding_mask)
+
+        sequence_output = keras.layers.LayerNormalization(
+            name="layer_norm",
+            axis=-1,
+            epsilon=1e-05,
+            dtype=tf.float32,
+        )(x)
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs={
+                "token_ids": token_ids,
+                "padding_mask": padding_mask,
+            },
+            outputs=sequence_output,
+            **kwargs,
+        )
+        # All references to `self` below this line
+        self.vocabulary_size = vocabulary_size
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.hidden_dim = hidden_dim
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
+        self.max_sequence_length = max_sequence_length
+        self.layer_norm_epsilon = layer_norm_epsilon
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocabulary_size": self.vocabulary_size,
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "hidden_dim": self.hidden_dim,
+                "intermediate_dim": self.intermediate_dim,
+                "dropout": self.dropout,
+                "max_sequence_length": self.max_sequence_length,
+                "layer_norm_epsilon": self.layer_norm_epsilon
+            }
+        )
+        return config
+
+    @property
+    def token_embedding(self):
+        return self.get_layer("token_embedding")
+
+    @classmethod
```

Comment:
we can probably leave this off for simplicity for now, this is easily a standalone follow up.

Reply:
I shall remove it in this PR